components/tensorboard-controller/controllers/tensorboard_controller.go

```diff
@@ -281,3 +328,102 @@ func isCloudPath(path string) bool {
 func isGoogleCloudPath(path string) bool {
 	return strings.HasPrefix(path, "gs://")
 }
+
+func isPVCPath(path string) bool {
+	return strings.HasPrefix(path, "pvc://")
+}
+
+func extractPVCName(path string) string {
+	trimmed := strings.TrimPrefix(path, "pvc://") //removing "pvc://" prefix
+	ending := strings.Index(trimmed, "/")         //finding ending index of pvc-name string
+	if ending == -1 {
+		return trimmed
+	} else {
+		return trimmed[0:ending]
+	}
+}
+
+func extractPVCSubPath(path string) string {
+	trimmed := strings.TrimPrefix(path, "pvc://") //removing "pvc://" prefix
+	start := strings.Index(trimmed, "/")          //finding starting index of local path inside PVC
+	if start == -1 || len(trimmed) == start+1 {
+		return ""
+	} else {
+		return trimmed[start+1:]
+	}
+}
+
+//Searches a corev1.PodList for running pods and returns
+//a running corev1.Pod (if exists)
+func findRunningPod(pods *corev1.PodList) corev1.Pod {
+	for _, pod := range pods.Items {
+		if pod.Status.Phase == "Running" {
+			return pod
+		}
+	}
+
+	return corev1.Pod{}
+}
+
+func extractNodeName(pod corev1.Pod) string {
+	return pod.Spec.NodeName
+}
+
+func generateNodeAffinity(affinity *corev1.Affinity, pvcname string, r *TensorboardReconciler, tb *tensorboardv1alpha1.Tensorboard) error {
+	var nodename string
+	var pods = &corev1.PodList{}
+	var pod corev1.Pod
+
+	//List all pods that access the PVC that has ClaimName: pvcname.
+	//NOTE: We use only one custom field selector to filter out pods that don't use this PVC.
+	if err := r.List(context.Background(), pods, client.InNamespace(tb.Namespace), client.MatchingFields{"spec.volumes.persistentvolumeclaim.claimname": pvcname}); err != nil {
+		return fmt.Errorf("List pods error: %v", err)
+	}
+
+	//Find a running pod that uses the PVC.
+	pod = findRunningPod(pods)
+
+	//If there is no running pod that uses the PVC, then: nodename == "".
+	//Else, nodename contains the name of the node that the pod is running on.
+	nodename = extractNodeName(pod)
+
+	//In this case, there is a running pod that uses the PVC, therefore we create
+	//a nodeAffinity field so that the Tensorboard server will be scheduled (if possible)
+	//on the same node as the running pod.
+	if nodename != "" {
```

Comment:
nit: You could have an `if nodename == "" {` here and return nil if that holds.

Then you could move the code below one tab to the left, outside of an if clause, to make it a little bit more simple to read